CRM/Core/BAO/Domain.php

```diff
@@ -366,7 +366,14 @@ public static function getDefaultReceiptFrom() {
    */
   public static function getNoReplyEmailAddress() {
     $emailDomain = CRM_Core_BAO_MailSettings::defaultDomain();
-    return "do-not-reply@$emailDomain";
+    $notReplyEmailAddress = \Civi::settings()->get('no_reply_email_address');
+
+    if (!empty($notReplyEmailAddress)) {
+      $notReplyEmailAddress = CRM_Core_OptionGroup::values('from_email_address', NULL, NULL, NULL, ' AND value = ' . $notReplyEmailAddress);
+      $notReplyEmailAddress = current($notReplyEmailAddress);
+    }
+
+    return $notReplyEmailAddress ?: "do-not-reply@$emailDomain";
```

Comment:
This is easier with the API:

```suggestion
    $noReplyAddress = \Civi\Api4\Setting::get(FALSE)
      ->addSelect('no_reply_email_address:label')
      ->execute()->first();

    return $noReplyAddress ?: "do-not-reply@$emailDomain";
```